components/Frame/HeaderNew.js

```diff
@@ -0,0 +1,387 @@
+import React, { useMemo, useState, useRef, useEffect } from 'react'
+import { css } from 'glamor'
+import { compose } from 'react-apollo'
+import { withRouter } from 'next/router'
+import {
+  Logo,
+  colors,
+  mediaQueries,
+  ColorContext,
+  HeaderHeightProvider
+} from '@project-r/styleguide'
+
+import { Router, matchPath } from '../../lib/routes'
+import { withMembership } from '../Auth/checkRoles'
+import withT from '../../lib/withT'
+import withInNativeApp, { postMessage } from '../../lib/withInNativeApp'
+import { shouldIgnoreClick } from '../Link/utils'
+import NotificationIconNew from '../Notifications/NotificationIconNew'
+import BackIcon from '../Icons/Back'
+
+import UserNew from './UserNew'
+import Popover from './Popover'
+import NavPopover from './Popover/NavNew'
+import UserNavPopover from './Popover/UserNav'
+import LoadingBar from './LoadingBar'
+import Pullable from './Pullable'
+import HLine from './HLine'
+import ToggleNew from './ToggleNew'
+import SecondaryNav from './SecondaryNav'
+
+import {
+  HEADER_HEIGHT,
+  HEADER_HEIGHT_MOBILE,
+  SUBHEADER_HEIGHT,
+  ZINDEX_POPOVER,
+  LOGO_WIDTH,
+  LOGO_PADDING,
+  LOGO_WIDTH_MOBILE,
+  LOGO_PADDING_MOBILE
+} from '../constants'
+
+const isActiveRoute = (active, route, params = {}) =>
+  !!active &&
+  active.route === route &&
+  Object.keys(params).every(key => active.params[key] === params[key])
+
+const isFront = router => {
+  const active = matchPath(router.asPath)
+  return isActiveRoute(active, 'index', {})
+}
+
+let routeChangeStarted
+
+const HeaderNew = ({
+  inNativeApp,
+  inNativeIOSApp,
+  isMember,
+  dark,
+  me,
+  t,
+  secondaryNav,
+  showSecondary,
+  router,
+  formatColor,
+  pullable = true,
+  hasOverviewNav,
+  children
+}) => {
+  const [isAnyNavExpanded, setIsAnyNavExpanded] = useState(false)
+  const [expandedNav, setExpandedNav] = useState(null)
+  const [isMobile, setIsMobile] = useState()
+  const [headerHeightState, setHeaderHeightState] = useState()
+  const [headerOffset, setHeaderOffset] = useState(0)
+  const [userNavExpanded, setUserNavExpanded] = useState(false)
+
+  const fixedRef = useRef()
+  const diff = useRef(0)
+  const lastY = useRef()
+  const headerHeight = useRef()
+  const lastDiff = useRef()
+
+  const textFill = dark ? colors.negative.text : colors.text
+  const logoFill = dark ? colors.logoDark || '#fff' : colors.logo || '#000'
+  const backButton = inNativeIOSApp && me && !isFront(router)
+
+  const toggleExpanded = target => {
+    if (target.id === expandedNav) {
```

Comment:
why go trough the DOM? Instead of passing `e.currentTarget` you could just directly pass the id / nav name?

Currently clicking on the profile pic when the user nav is already open does not hide it. Some logic here is broken.

Reply:
Fixed the DOM part. The case of the nav not closing on clicking the user icon when the user nav is already open is as intended. Both navs close via the close button on the right. I thought it could be perceived as inconsistent if suddenly the user icon would double as a close button, even though there is an explicit close button on the right.